Prevent bench states discovering addresses of the wrong kind

Anviking opened the pull request:
- [x] Prevent bench states discovering addresses of the wrong kind

### Comments

https://buildkite.com/input-output-hk/cardano-wallet-nightly/builds/1822#0188762c-180e-4765-925a-c526966d5672

Last succeeding benchmark for comparison: https://buildkite.com/input-output-hk/cardano-wallet-nightly/builds/1781#0187c020-03e3-48ea-910b-f5ad32d7af66

### Issue Number

ADP-3055

Anviking committed:
Prevent bench states discovering addresses of the wrong kind

jonathanknowles commented on lib/wallet/src/Cardano/Wallet/Address/Discovery/Sequential.hs:
```diff
@@ -734,6 +738,16 @@ instance KnownNat p => IsOurs (SeqAnyState n k p) Address where
+            Ledger.Addr _net (Ledger.ScriptHashObj _) _ -> True
+            Ledger.AddrBootstrap _ -> False
+
+
```
```suggestion

```

jonathanknowles commented on lib/wallet/src/Cardano/Wallet/Address/Discovery/Sequential.hs:
```diff
@@ -734,6 +738,16 @@ instance KnownNat p => IsOurs (SeqAnyState n k p) Address where
         double :: Integral a => a -> Double
         double = fromIntegral
 
+        -- ADP-3056: Consider making this implementation a part of a
+        -- @validate :: UTxOAssumptions -> Address -> Bool@ function
+        -- somewhere in the Write module hierarchy.
```
👍🏻

Anviking committed:
Increase rnd-bench ownership from 0.01% to 0.1%

As 0.01% of all bootstrap addresses is much smaller than 0.01% of all
addresses, we raise it to 0.1% to compensate.
